File: examples/wrappers_example/test_mathlib.go
package main

import (
	"fmt"
	"mathlib"
)

func main() {
	a := mathlib.Vector{X: 1.0, Y: 2.0, Z: 3.0}
	b := mathlib.Vector{X: 4.0, Y: 5.0, Z: 6.0}
	var r mathlib.Vector

	fmt.Println("=== Vector Math Library Test (Go) ===")
	fmt.Println()

	mathlib.Vec_add(&a, &b, &r)
	fmt.Printf("a + b         = (%.1f, %.1f, %.1f)\n", r.X, r.Y, r.Z)

	fmt.Printf("length(a)     = %.4f\n", mathlib.Vec_length(&a))
	fmt.Printf("length(b)     = %.4f\n", mathlib.Vec_length(&b))
	fmt.Printf("dot(a, b)     = %.1f\n", mathlib.Vec_dot(&a, &b))

	mathlib.Vec_scale(&a, 2.0)
	fmt.Printf("a * 2         = (%.1f, %.1f, %.1f)\n", a.X, a.Y, a.Z)

	fmt.Println()
	fmt.Println("All tests passed!")
}

File: examples/wrappers_example/test_mathlib.rs
use mathlib::*;

fn main() {
    let a = Vector { x: 1.0, y: 2.0, z: 3.0 };
    let b = Vector { x: 4.0, y: 5.0, z: 6.0 };
    let mut r = unsafe { std::mem::zeroed() };

    println!("=== Vector Math Library Test (Rust) ===");
    println!();

    unsafe {
        vec_add(&a, &b, &mut r);
        println!("a + b         = ({:.1}, {:.1}, {:.1})", r.x, r.y, r.z);
        println!("length(a)     = {:.4}", vec_length(&a));
        println!("length(b)     = {:.4}", vec_length(&b));
        println!("dot(a, b)     = {:.1}", vec_dot(&a, &b));
    }

    println!();
    println!("All tests passed!");
}
